Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UnitData.xml

```diff
@@ -7249,25 +7249,37 @@
-            <LayoutButtons Face="AP_HireHammerSecurities" Type="AbilCmd" AbilCmd="AP_SummonMercenaries,3" Row="0" Column="2"/>
-            <LayoutButtons Face="AP_HireSpartanCompany" Type="AbilCmd" AbilCmd="AP_SummonMercenaries,2" Row="0" Column="3"/>
-            <LayoutButtons Face="AP_HireMercMedic" Type="AbilCmd" AbilCmd="AP_SummonMercenaries,8" Row="0" Column="4"/>
-            <LayoutButtons Face="AP_HireSiegeBreakers" Type="AbilCmd" AbilCmd="AP_SummonMercenaries,4" Row="1" Column="0"/>
-            <LayoutButtons Face="AP_HireHelsAngels" Type="AbilCmd" AbilCmd="AP_SummonMercenaries,5" Row="1" Column="1"/>
-            <LayoutButtons Face="AP_HireDuskWing" Type="AbilCmd" AbilCmd="AP_SummonMercenaries,6" Row="1" Column="2"/>
-            <LayoutButtons Face="AP_HireMercReaper" Type="AbilCmd" AbilCmd="AP_SummonMercenaries,9" Row="1" Column="3"/>
-            <LayoutButtons Face="SelectBuilder" Type="SelectBuilder" AbilCmd="255,3" Row="1" Column="3"/>
+            <LayoutButtons Face="AP_HireKelmorianMiners" Type="AbilCmd" AbilCmd="AP_SummonMercenaries,Train1" Row="0" Column="0"/>
+            <LayoutButtons Face="AP_HireDevilDogs" Type="AbilCmd" AbilCmd="AP_SummonMercenaries,Train2" Row="0" Column="1"/>
+            <LayoutButtons Face="AP_HireHammerSecurities" Type="AbilCmd" AbilCmd="AP_SummonMercenaries,Train4" Row="0" Column="2"/>
+            <LayoutButtons Face="AP_HireMercReaper" Type="AbilCmd" AbilCmd="AP_SummonMercenaries,Train10" Row="0" Column="3"/>
+            <LayoutButtons Face="AP_HireMercMedic" Type="AbilCmd" AbilCmd="AP_SummonMercenaries,Train9" Row="0" Column="4"/>
+            <LayoutButtons Face="SelectBuilder" Type="SelectBuilder" AbilCmd="255,255" Row="1" Column="3"/>
             <LayoutButtons Face="Rally" Type="AbilCmd" AbilCmd="Rally,Rally1" Row="1" Column="4"/>
-            <LayoutButtons Face="AP_HireDukesRevenge" Type="AbilCmd" AbilCmd="AP_SummonMercenaries,7" Row="2" Column="0"/>
-            <LayoutButtons Face="AP_HireMercWraith" Type="AbilCmd" AbilCmd="AP_SummonMercenaries,10" Row="2" Column="1"/>
-            <LayoutButtons Face="AP_HireMercLiberator" Type="AbilCmd" AbilCmd="AP_SummonMercenaries,11" Row="2" Column="2"/>
-            <LayoutButtons Face="AP_HireBrynhildFighter" Type="AbilCmd" AbilCmd="AP_SummonMercenaries,12" Row="2" Column="3"/>
-            <LayoutButtons Face="AP_HireMercThor" Type="AbilCmd" AbilCmd="AP_SummonMercenaries,13" Row="2" Column="4"/>
+            <LayoutButtons Face="AP_MercVehicle" Type="Submenu" SubmenuCardId="0002" Row="2" Column="1"/>
+            <LayoutButtons Face="AP_MercShip" Type="Submenu" SubmenuCardId="0003" Row="2" Column="2"/>
             <LayoutButtons Face="Halt" Type="AbilCmd" AbilCmd="BuildInProgress,Halt" Row="2" Column="3"/>
             <LayoutButtons Face="CancelBuilding" Type="AbilCmd" AbilCmd="BuildInProgress,Cancel" Row="2" Column="4"/>
         </CardLayouts>
+        <CardLayouts CardId="0002">

```

Comment:
Right now, if you select any vehicle or ship merc, the submenu will revert back to the infantry tab. You can set a flag on the submenus ("sticky" I think?) to make it stay on the submenu when you use an ability. It will reset if you deselect+reselect the building. Not sure what behaviour you think would work best.

Reply:
My bad. I'll get it fixed

Reply:
I wouldn't call it a bad, I was just pointing out another direction you could take the design. Either could work.

Reply:
Before Phan's changes to Protoss building there was behavior-driven stuff instead using submenus. It worked in a way that it looked like submenus but was sticky across re-selecting.

The implementation of this trick relies on behaviors and having requirements for each group that require behavior on the unit. The tab selection button enables the behavior so the correct options are shown.

I think it could be the desired behavior there as I set Vehicles and it stays so then across re-selecting

Reply:
A second commit was done to make it behave as desired.
Now if I call a mercenary from a submenu, the submenu will not change.

Reply:
re: Behaviour-based things, that was quite a complicated system (the new system is also complicated), and I don't think that level of complexity is necessary here. The fastest default merc cooldowns are 5 minutes, (2m30 with cooldown speed, 4minutes with proposed balance changes in chat), that's slow enough that I think it's reasonable to select the submenu every time you're calling something down.

Reply:
The merc compound can be used as an emergency base defense tool. If the submenu defaults to infantry after every calldown, it may disorient some players in stressful situations.

Reply:
it should be left on the last visited tab and not default to an empty tab if possible on start

Reply:
Pretty sure there is no empty tab, it will default to infantry.